pkg/network/usm/istio.go

```diff
@@ -285,6 +287,10 @@ func (m *istioMonitor) getEnvoyPath(pid uint32) string {
 	}
 
 	buffer = buffer[:n]
+
+	if m.envoyPath != "" {
+		envoyCmd = []byte(m.envoyPath)
+	}
```

Comment:
you should do that in the constructor, and do it only once
if the config is empty, then set the default option into `m.envoyPath`